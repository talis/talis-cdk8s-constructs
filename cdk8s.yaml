language: typescript
imports:
  - k8s@1.30.0
