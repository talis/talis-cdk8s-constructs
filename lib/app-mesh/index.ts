export * from "./ingress-gateway-nlb";
export * from "./app-service";
