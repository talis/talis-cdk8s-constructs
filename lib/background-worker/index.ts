export * from "./background-worker-props";
export * from "./background-worker";
