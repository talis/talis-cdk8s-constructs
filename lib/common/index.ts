export * from "./container-props";
export * from "./env-util";
export * from "./name-util";
export * from "./statefulset-util";
export * from "./value-util";
export * from "./workload-props";
