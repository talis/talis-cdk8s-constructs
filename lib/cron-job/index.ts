export * from "./cron-job-props";
export * from "./cron-job";
