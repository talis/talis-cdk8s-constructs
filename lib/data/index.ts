export * from "./config-map";
export * from "./secret";
export * from "./factory";
