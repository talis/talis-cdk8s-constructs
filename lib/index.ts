export * from "./cron-job";
export * from "./background-worker";
export * from "./data";
export * from "./job";
export * from "./redis";
export * from "./talis-chart";
export * from "./web-service";
