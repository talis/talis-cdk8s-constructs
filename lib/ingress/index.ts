export * from "./ingress-props";
export * from "./ingress";
