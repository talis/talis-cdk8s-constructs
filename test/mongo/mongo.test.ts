import { Chart, Testing } from "cdk8s";
import { KubeService } from "../../imports/k8s";
import { Mongo, MongoProps } from "../../lib";
import { makeChart } from "../test-util";

const requiredProps = {
  release: "v1",
};

function synthMongo(props: MongoProps = requiredProps) {
  const chart = Testing.chart();
  new Mongo(chart, "mongo-test", props);
  const results = Testing.synth(chart);
  return results;
}

describe("Mongo", () => {
  describe("Props", () => {
    test("Minimal required props", () => {
      const chart = Testing.chart();
      new Mongo(chart, "mongo-test", requiredProps);
      const results = Testing.synth(chart);
      expect(results).toMatchSnapshot();
    });

    test("All the props", () => {
      const app = Testing.app();
      const chart = new Chart(app, "test", {
        namespace: "test",
        labels: {
          app: "my-app",
          environment: "test",
          region: "local",
        },
      });
      const selectorLabels = {
        app: "my-app",
        role: "mongo",
        instance: "test",
      };

      new Mongo(chart, "mongo-test", {
        ...requiredProps,
        selectorLabels,
      });
      const results = Testing.synth(chart);
      expect(results).toMatchSnapshot();
    });
  });

  describe("Service instance", () => {
    test("Exposes service object through property", () => {
      const chart = makeChart();
      const mongo = new Mongo(chart, "mongo-test", requiredProps);
      expect(mongo.service).toBeDefined();
      expect(mongo.service).toBeInstanceOf(KubeService);
      expect(mongo.service.name).toEqual("mongo-test");
    });
  });

  describe("Container release", () => {
    test("Default container release", () => {
      const results = synthMongo();
      const mongo = results.find((obj) => obj.kind === "StatefulSet");
      expect(mongo).toHaveProperty(
        "spec.template.spec.containers[0].image",
        "mongo:v1"
      );
    });

    test("Container release set explicitly", () => {
      const results = synthMongo({
        ...requiredProps,
        release: "12345",
      });
      const mongo = results.find((obj) => obj.kind === "StatefulSet");
      expect(mongo).toHaveProperty(
        "spec.template.spec.containers[0].image",
        "mongo:12345"
      );
    });
  });

  describe("Storage Engine", () => {
    test("Default storage engine", () => {
      const results = synthMongo({
        ...requiredProps,
      });
      const mongo = results.find((obj) => obj.kind === "StatefulSet");
      expect(mongo).toHaveProperty("spec.template.spec.containers[0].command", [
        "--smallfiles",
        "--storageEngine",
        "mmapv1",
      ]);
    });

    test("Storage engine set explicitly", () => {
      const results = synthMongo({
        ...requiredProps,
        storageEngine: "wiredTiger",
      });
      const mongo = results.find((obj) => obj.kind === "StatefulSet");
      expect(mongo).toHaveProperty("spec.template.spec.containers[0].command", [
        "--smallfiles",
        "--storageEngine",
        "wiredTiger",
      ]);
    });
  });
});
