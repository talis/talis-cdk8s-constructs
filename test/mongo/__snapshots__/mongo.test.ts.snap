// Vitest Snapshot v1

exports[`Mongo > Props > All the props 1`] = `
[
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "labels": {
        "app": "my-app",
        "environment": "test",
        "instance": "test",
        "region": "local",
        "release": "v1",
        "role": "mongo",
      },
      "name": "test-mongo-test-c844268f",
      "namespace": "test",
    },
    "spec": {
      "clusterIP": "None",
      "ports": [
        {
          "port": 27017,
          "protocol": "TCP",
        },
      ],
      "selector": {
        "app": "my-app",
        "instance": "test",
        "role": "mongo",
      },
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "StatefulSet",
    "metadata": {
      "labels": {
        "app": "my-app",
        "environment": "test",
        "instance": "test",
        "region": "local",
        "release": "v1",
        "role": "mongo",
      },
      "name": "test-mongo-test-mongo-test-sts-c8cbd639",
      "namespace": "test",
    },
    "spec": {
      "replicas": 1,
      "selector": {
        "matchLabels": {
          "app": "my-app",
          "instance": "test",
          "role": "mongo",
        },
      },
      "serviceName": "test-mongo-test-c844268f",
      "template": {
        "metadata": {
          "labels": {
            "app": "my-app",
            "environment": "test",
            "instance": "test",
            "region": "local",
            "release": "v1",
            "role": "mongo",
          },
        },
        "spec": {
          "containers": [
            {
              "args": [
                "--storageEngine",
                "wiredTiger",
              ],
              "image": "mongo:v1",
              "livenessProbe": {
                "failureThreshold": 5,
                "initialDelaySeconds": 5,
                "tcpSocket": {
                  "port": 27017,
                },
                "timeoutSeconds": 5,
              },
              "name": "mongo",
              "ports": [
                {
                  "containerPort": 27017,
                },
              ],
              "readinessProbe": {
                "exec": {
                  "command": [
                    "mongo",
                    "--eval",
                    "db.adminCommand('ping')",
                  ],
                },
                "failureThreshold": 5,
                "initialDelaySeconds": 5,
                "timeoutSeconds": 5,
              },
              "resources": {
                "limits": {
                  "cpu": "100m",
                  "memory": "500Mi",
                },
              },
              "volumeMounts": [
                {
                  "mountPath": "/data/db",
                  "name": "mongo-data",
                },
              ],
            },
          ],
        },
      },
      "volumeClaimTemplates": [
        {
          "metadata": {
            "name": "mongo-data",
          },
          "spec": {
            "accessModes": [
              "ReadWriteOnce",
            ],
            "resources": {
              "requests": {
                "storage": "20Gi",
              },
            },
            "storageClassName": "general-purpose-delete",
          },
        },
      ],
    },
  },
]
`;

exports[`Mongo > Props > Minimal required props 1`] = `
[
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "labels": {
        "instance": "mongo-test",
        "release": "v1",
        "role": "mongo",
      },
      "name": "test-mongo-test-c844268f",
    },
    "spec": {
      "clusterIP": "None",
      "ports": [
        {
          "port": 27017,
          "protocol": "TCP",
        },
      ],
      "selector": {
        "instance": "mongo-test",
        "role": "mongo",
      },
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "StatefulSet",
    "metadata": {
      "labels": {
        "instance": "mongo-test",
        "release": "v1",
        "role": "mongo",
      },
      "name": "test-mongo-test-mongo-test-sts-c8cbd639",
    },
    "spec": {
      "replicas": 1,
      "selector": {
        "matchLabels": {
          "instance": "mongo-test",
          "role": "mongo",
        },
      },
      "serviceName": "test-mongo-test-c844268f",
      "template": {
        "metadata": {
          "labels": {
            "instance": "mongo-test",
            "release": "v1",
            "role": "mongo",
          },
        },
        "spec": {
          "containers": [
            {
              "args": [
                "--storageEngine",
                "mmapv1",
                "--smallfiles",
              ],
              "image": "mongo:v1",
              "livenessProbe": {
                "failureThreshold": 5,
                "initialDelaySeconds": 5,
                "tcpSocket": {
                  "port": 27017,
                },
                "timeoutSeconds": 5,
              },
              "name": "mongo",
              "ports": [
                {
                  "containerPort": 27017,
                },
              ],
              "readinessProbe": {
                "exec": {
                  "command": [
                    "mongo",
                    "--eval",
                    "db.adminCommand('ping')",
                  ],
                },
                "failureThreshold": 5,
                "initialDelaySeconds": 5,
                "timeoutSeconds": 5,
              },
              "resources": {
                "limits": {
                  "cpu": "100m",
                  "memory": "500Mi",
                },
              },
              "volumeMounts": [
                {
                  "mountPath": "/data/db",
                  "name": "mongo-data",
                },
              ],
            },
          ],
        },
      },
      "volumeClaimTemplates": [
        {
          "metadata": {
            "name": "mongo-data",
          },
          "spec": {
            "accessModes": [
              "ReadWriteOnce",
            ],
            "resources": {
              "requests": {
                "storage": "20Gi",
              },
            },
            "storageClassName": "general-purpose-delete",
          },
        },
      ],
    },
  },
]
`;
